sound/soc/sof/topology.c

```diff
@@ -1738,63 +1738,18 @@ static int sof_process_load(struct snd_soc_component *scomp, int index,
 	struct sof_abi_hdr *pdata = NULL;
 	struct sof_ipc_comp_process *process;
 	size_t ipc_size, ipc_data_size = 0;
-	int ret, i, offset = 0;
+	int ret, i;
+	int ipc_cmd, ctrl_type;
 
 	if (type == SOF_COMP_NONE) {
 		dev_err(sdev->dev, "error: invalid process comp type %d\n",
 			type);
 		return -EINVAL;
 	}
 
-	/*
-	 * get possible component controls - get size of all pdata,
-	 * then memcpy with headers
-	 */
-	for (i = 0; i < widget->num_kcontrols; i++) {
-
-		kc = &widget->kcontrol_news[i];
-
-		switch (widget->dobj.widget.kcontrol_type) {
-		case SND_SOC_TPLG_TYPE_MIXER:
-			sm = (struct soc_mixer_control *)kc->private_value;
-			scontrol = sm->dobj.private;
-			break;
-		case SND_SOC_TPLG_TYPE_BYTES:
-			sbe = (struct soc_bytes_ext *)kc->private_value;
-			scontrol = sbe->dobj.private;
-			break;
-		case SND_SOC_TPLG_TYPE_ENUM:
-			se = (struct soc_enum *)kc->private_value;
-			scontrol = se->dobj.private;
-			break;
-		default:
-			dev_err(sdev->dev, "error: unknown kcontrol type %d in widget %s\n",
-				widget->dobj.widget.kcontrol_type,
-				widget->name);
-			return -EINVAL;
-		}
-
-		if (!scontrol) {
-			dev_err(sdev->dev, "error: no scontrol for widget %s\n",
-				widget->name);
-			return -EINVAL;
-		}
-
-		/* don't include if no private data */
-		pdata = scontrol->control_data->data;
-		if (!pdata)
-			continue;
-
-		/* make sure data is valid - data can be updated at runtime */
-		if (pdata->magic != SOF_ABI_MAGIC)
-			continue;
-
-		ipc_data_size += pdata->size;
-	}
-
 	ipc_size = sizeof(struct sof_ipc_comp_process) +
 		le32_to_cpu(private->size) +
-		ipc_data_size;
+	  ipc_data_size;
```

Comment:
I don't get at all what this simplification does, we now have ipc_data_size = 0?

Reply:
removed totally.